PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/mysql/creation/MySQLCreationBasic.java

```diff
@@ -0,0 +1,128 @@
+ *
+ * The above copyright notice and this permission notice shall be included in all copies or substantial portions of
+ * the Software.
+ *
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
+ * THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.microsoft.azure.toolkit.intellij.mysql.creation;
+
+import com.microsoft.azure.toolkit.intellij.common.AzureFormPanel;
+import com.microsoft.azure.toolkit.intellij.common.AzurePasswordFieldInput;
+import com.microsoft.azure.toolkit.intellij.common.TextDocumentListenerAdapter;
+import com.microsoft.azure.toolkit.intellij.mysql.AdminUsernameTextField;
+import com.microsoft.azure.toolkit.intellij.mysql.ServerNameTextField;
+import com.microsoft.azure.toolkit.lib.common.form.AzureFormInput;
+import com.microsoft.azure.toolkit.lib.mysql.AzureMySQLConfig;
+import lombok.Getter;
+import org.apache.commons.lang3.StringUtils;
+
+import javax.swing.*;
+import javax.swing.event.DocumentListener;
+import java.util.Arrays;
+import java.util.List;
+
+public class MySQLCreationBasic extends JPanel implements AzureFormPanel<AzureMySQLConfig> {
+
+    private JPanel rootPanel;
+    @Getter
+    private ServerNameTextField serverNameTextField;
+    @Getter
+    private AdminUsernameTextField adminUsernameTextField;
+    @Getter
+    private JPasswordField passwordField;
+    @Getter
+    private JPasswordField confirmPasswordField;
+
+    private AzurePasswordFieldInput passwordFieldInput;
+    private AzurePasswordFieldInput confirmPasswordFieldInput;
+
+    private AzureMySQLConfig config;
+
+    MySQLCreationBasic(AzureMySQLConfig config) {
+        super();
+        this.config = config;
+        $$$setupUI$$$(); // tell IntelliJ to call createUIComponents() here.
+        init();
+        initListeners();
+        setData(config);
+    }
+
+    private void init() {
+        passwordFieldInput = PasswordUtils.generatePasswordFieldInput(this.passwordField, this.adminUsernameTextField);
+        confirmPasswordFieldInput = PasswordUtils.generateConfirmPasswordFieldInput(this.confirmPasswordField, this.passwordField);
+    }
+
+    private void initListeners() {
+        this.adminUsernameTextField.getDocument().addDocumentListener(generateAdminUsernameListener());
+    }
+
+    private DocumentListener generateAdminUsernameListener() {
+        return new TextDocumentListenerAdapter() {
+            @Override
+            public void onDocumentChanged() {
+                if (!adminUsernameTextField.isPasswordInitialized()) {
+                    adminUsernameTextField.setPasswordInitialized(true);
```

Comment:
why user name has to know whether password is initialized?

Reply:
@Flanker32 also mentioned it. and I already updated it now.